src/arithmetic/arithmetic_expression_construct.c

```diff
@@ -441,6 +443,96 @@ static AR_ExpNode *_AR_ExpFromNamedPath(const cypher_astnode_t *path) {
 	return op;
 }
 
+static AR_ExpNode *_AR_ExpFromShortestPath(const cypher_astnode_t *path) {
```

Comment:
Are we validating that both the source and destination nodes are aliased ?
in addition, do we make sure both source and destination nodes are considered as "referenced" so we won't lose them as in: `MATCH (a)-[]->(b)-[]->(c), (d) RETURN shortestPath(b, d)`

Reply:
Good catch, I will fix this.